syntax = "proto2";

message RegisterNodeRequest {
	required int32 version = 1;
}

message RegisterNodeResponse {
	required int32 code = 1;
}

message BootstrapRequest {
	required string url = 1;
	required string rev = 2;
	required string opt = 3;
}

message BootstrapResponse {
	enum Code {
		OK = 0;
		EURL = 1;
		EREV = 2;
		ECFG = 3;
	}

	required Code code = 1;
}

message CompileRequest {
	required string files = 1;
}

message CompileResponse {
	required string file = 1;

	// Send compilation error string or file contents
	required string error = 2;
	required bytes data = 3;
}
